lib/datura/elasticsearch/index.rb

```diff
@@ -100,4 +112,21 @@ def set_schema
     }
   end
 
+  # doc: ruby hash corresponding with Elasticsearch document JSON
+  def valid_document?(doc)
+    get_schema_mapping if @schema_mapping.nil?
```

Comment:
I believe the check for `@schema_mapping` being nil is already present inside `get_schema_mapping` so I think that conditional can be dropped on this line.

I saw this was updated to `!defined?` but I think the same suggestion applies.

Reply:
removed